sdk/storage/README.md

```diff
@@ -47,7 +47,7 @@ For general suggestions about Azure, use our [Azure feedback forum](https://feed
 On Windows, dependencies are managed by [vcpkg](https://github.com/microsoft/vcpkg). You can reference the [Quick Start](https://github.com/microsoft/vcpkg#quick-start-windows) to quickly set yourself up.
 After Vcpkg is initialized and bootstrapped, you can install the dependencies:
 ```BatchFile
-vcpkg.exe install libxml2:x64-windows curl:x64-windows
+vcpkg.exe install libxml2:x64-windows-static curl:x64-windows-static
```

Comment:
What's the motivation of this change? What kind of problem does it solve?

Reply:
It solves the problem where default build settings won't work with this readme.

Reply:
I doubt that. Can you elaborate the environment to reproduce this issue?

Reply:
No need for that, check this line: https://github.com/Azure/azure-sdk-for-cpp/blob/master/CMakeSettings.json#L18

Reply:
How does that prove `default build settings won't work` as you said?

Reply:
Default build setting requires `static` libraries as dependencies, if we do not specify that in vcpkg installation command, the packages will by default prompt not found.

Reply:
I'm sorry but it seems to me that you've been sidestepping the question. 

> Default build setting requires static libraries as dependencies, if we do not specify that in vcpkg installation command, the packages will by default prompt not found.

This is just your claim without any evidence to back it up.

```
C:\Users\jinmhu.FAREAST\Desktop>git clone https://github.com/microsoft/vcpkg.git
Cloning into 'vcpkg'...
remote: Enumerating objects: 94316, done.
remote: Total 94316 (delta 0), reused 0 (delta 0), pack-reused 94316 eceiving objects: 100% (94316/94316), 27.67 MiB |
Receiving objects: 100% (94316/94316), 28.21 MiB | 1.25 MiB/s, done.
Resolving deltas: 100% (58359/58359), done.
Updating files: 100% (5611/5611), done.

C:\Users\jinmhu.FAREAST\Desktop>cd vcpkg

C:\Users\jinmhu.FAREAST\Desktop\vcpkg>bootstrap-vcpkg.bat

Building vcpkg.exe ...

  Using triplet "x86-windows" from "C:\ProgramFiles\vcpkg\installed\x86-windows\"
  The Vcpkg manifest was disabled, but we found a manifest file in \. You may want to enable vcpkg manifests in your pr
  operties page or pass /p:VcpkgEnableManifest=true to the msbuild invocation.
  pch.cpp
  archives.cpp
  checks.cpp
  chrono.cpp
  cofffilereader.cpp
  downloads.cpp
  enums.cpp
  files.cpp
  hash.cpp
  json.cpp
  machinetype.cpp
  parse.cpp
  strings.cpp
  stringview.cpp
  system.cpp
  system.print.cpp
  system.process.cpp
  unicode.cpp
  binarycaching.cpp
  binaryparagraph.cpp
  build.cpp
  buildenvironment.cpp
  cmakevars.cpp
  commands.autocomplete.cpp
  commands.buildexternal.cpp
  commands.cache.cpp
  commands.ci.cpp
  commands.ciclean.cpp
  commands.contact.cpp
  commands.cpp
  commands.create.cpp
  commands.dependinfo.cpp
  commands.edit.cpp
  commands.env.cpp
  commands.fetch.cpp
  commands.format-manifest.cpp
  commands.hash.cpp
  commands.info.cpp
  commands.integrate.cpp
  commands.list.cpp
  commands.owns.cpp
  commands.porthistory.cpp
  commands.portsdiff.cpp
  commands.search.cpp
  commands.setinstalled.cpp
  commands.upgrade.cpp
  commands.upload-metrics.cpp
  commands.version.cpp
  commands.xvsinstances.cpp
  configuration.cpp
  dependencies.cpp
  export.cpp
  export.chocolatey.cpp
  export.ifw.cpp
  export.prefab.cpp
  globalstate.cpp
  help.cpp
  input.cpp
  install.cpp
  platform-expression.cpp
  metrics.cpp
  packagespec.cpp
  paragraphs.cpp
  portfileprovider.cpp
  postbuildlint.buildtype.cpp
  postbuildlint.cpp
  registries.cpp
  remove.cpp
  sourceparagraph.cpp
  statusparagraph.cpp
  statusparagraphs.cpp
  tools.cpp
  triplet.cpp
  update.cpp
  userconfig.cpp
  vcpkgcmdarguments.cpp
  vcpkglib.cpp
  vcpkgpaths.cpp
  versiont.cpp
  visualstudio.cpp
  vcpkg.cpp
  Generating code
  Previous IPDB not found, fall back to full compilation.
  All 20808 functions were compiled because no usable IPDB/IOBJ from previous compilation was found.
  Finished generating code
  vcpkg.vcxproj -> C:\Users\jinmhu.FAREAST\Desktop\vcpkg\toolsrc\windows-bootstrap\msbuild.x86.release\vcpkg.exe

Building vcpkg.exe... done.

Telemetry
---------
vcpkg collects usage data in order to help us improve your experience.
The data collected by Microsoft is anonymous.
You can opt-out of telemetry by re-running the bootstrap-vcpkg script with -disableMetrics,
passing --disable-metrics to vcpkg on the command line,
or by setting the VCPKG_DISABLE_METRICS environment variable.

Read more about vcpkg telemetry at docs/about/privacy.md

C:\Users\jinmhu.FAREAST\Desktop\vcpkg>vcpkg.exe install libxml2:x64-windows curl:x64-windows
Your feedback is important to improve Vcpkg! Please take 3 minutes to complete our survey by running: vcpkg contact --survey
Computing installation plan...
A suitable version of cmake was not found (required v3.18.4). Downloading portable cmake v3.18.4...
Downloading cmake...
  https://github.com/Kitware/CMake/releases/download/v3.18.4/cmake-3.18.4-win32-x86.zip -> C:\Users\jinmhu.FAREAST\Desktop\vcpkg\downloads\cmake-3.18.4-win32-x86.zip
Extracting cmake...
A suitable version of 7zip was not found (required v18.1.0). Downloading portable 7zip v18.1.0...
Downloading 7zip...
  https://www.nuget.org/api/v2/package/7-Zip.CommandLine/18.1.0 -> C:\Users\jinmhu.FAREAST\Desktop\vcpkg\downloads\7-zip.commandline.18.1.0.nupkg
Extracting 7zip...
A suitable version of nuget was not found (required v5.5.1). Downloading portable nuget v5.5.1...
Downloading nuget...
  https://dist.nuget.org/win-x86-commandline/v5.5.1/nuget.exe -> C:\Users\jinmhu.FAREAST\Desktop\vcpkg\downloads\22ea847d-nuget.exe
The following packages will be built and installed:
    curl[core,non-http,schannel,ssl,sspi,winssl]:x64-windows
  * libiconv[core]:x64-windows
  * liblzma[core]:x64-windows
    libxml2[core]:x64-windows
  * zlib[core]:x64-windows
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-windows...
A suitable version of powershell-core was not found (required v7.0.3). Downloading portable powershell-core v7.0.3...
Downloading powershell-core...
  https://github.com/PowerShell/PowerShell/releases/download/v7.0.3/PowerShell-7.0.3-win-x86.zip -> C:\Users\jinmhu.FAREAST\Desktop\vcpkg\downloads\PowerShell-7.0.3-win-x86.zip
Extracting powershell-core...
Starting package 1/5: zlib:x64-windows
Building package zlib[core]:x64-windows...
Could not locate cached archive: C:\Users\jinmhu.FAREAST\AppData\Local\vcpkg\archives\38\38209fa61175fa0cad1650a3b8ce192d5837f6df.zip
-- Downloading http://www.zlib.net/zlib-1.2.11.tar.gz...
-- Extracting source C:/Users/jinmhu.FAREAST/Desktop/vcpkg/downloads/zlib1211.tar.gz
-- Applying patch cmake_dont_build_more_than_needed.patch
-- Applying patch 0001-Prevent-invalid-inclusions-when-HAVE_-is-set-to-0.patch
-- Applying patch add_debug_postfix_on_mingw.patch
-- Using source at C:/Users/jinmhu.FAREAST/Desktop/vcpkg/buildtrees/zlib/src/1.2.11-63309e48e5.clean
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
-- Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-pkg-config-0.29.2-1-any.pkg.tar.xz...
-- Downloading https://repo.msys2.org/mingw/i686/mingw-w64-i686-libwinpthread-git-8.0.0.5906.c9a21571-1-any.pkg.tar.zst...
-- Using msys root at C:/Users/jinmhu.FAREAST/Desktop/vcpkg/downloads/tools/msys2/c809757c94447846
-- Fixing pkgconfig file: C:/Users/jinmhu.FAREAST/Desktop/vcpkg/packages/zlib_x64-windows/lib/pkgconfig/zlib.pc
-- Fixing pkgconfig file: C:/Users/jinmhu.FAREAST/Desktop/vcpkg/packages/zlib_x64-windows/debug/lib/pkgconfig/zlib.pc
-- Installing: C:/Users/jinmhu.FAREAST/Desktop/vcpkg/packages/zlib_x64-windows/share/zlib/copyright
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: C:\Users\jinmhu.FAREAST\AppData\Local\vcpkg\archives\38\38209fa61175fa0cad1650a3b8ce192d5837f6df.zip
Building package zlib[core]:x64-windows... done
Installing package zlib[core]:x64-windows...
Installing package zlib[core]:x64-windows... done
Elapsed time for package zlib:x64-windows: 18.31 s
Starting package 2/5: curl:x64-windows
Building package curl[core,non-http,schannel,ssl,sspi,winssl]:x64-windows...
Could not locate cached archive: C:\Users\jinmhu.FAREAST\AppData\Local\vcpkg\archives\bb\bb261c41eed293088ed03f299fd35cd42ad23936.zip
-- Downloading https://github.com/curl/curl/archive/315ee3fe75dade912b48a21ceec9ccda0230d937.tar.gz...
-- Extracting source C:/Users/jinmhu.FAREAST/Desktop/vcpkg/downloads/curl-curl-315ee3fe75dade912b48a21ceec9ccda0230d937.tar.gz
-- Applying patch 0002_fix_uwp.patch
-- Applying patch 0004_nghttp2_staticlib.patch
-- Applying patch 0005_remove_imp_suffix.patch
-- Applying patch 0006_fix_tool_depends.patch
-- Applying patch 0007_disable_tool_export_curl_target.patch
-- Applying patch 0009_fix_openssl_config.patch
-- Applying patch 0010_fix_othertests_cmake.patch
-- Applying patch 0011_fix_static_build.patch
-- Using source at C:/Users/jinmhu.FAREAST/Desktop/vcpkg/buildtrees/curl/src/da0230d937-b280319101.clean
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
-- Installing: C:/Users/jinmhu.FAREAST/Desktop/vcpkg/packages/curl_x64-windows/share/curl/curl-config
-- Installing: C:/Users/jinmhu.FAREAST/Desktop/vcpkg/packages/curl_x64-windows/share/curl/vcpkg-cmake-wrapper.cmake
-- Installing: C:/Users/jinmhu.FAREAST/Desktop/vcpkg/packages/curl_x64-windows/share/curl/copyright
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: C:\Users\jinmhu.FAREAST\AppData\Local\vcpkg\archives\bb\bb261c41eed293088ed03f299fd35cd42ad23936.zip
Building package curl[core,non-http,schannel,ssl,sspi,winssl]:x64-windows... done
Installing package curl[core,non-http,schannel,ssl,sspi,winssl]:x64-windows...
Installing package curl[core,non-http,schannel,ssl,sspi,winssl]:x64-windows... done
Elapsed time for package curl:x64-windows: 1.294 min
Starting package 3/5: libiconv:x64-windows
Building package libiconv[core]:x64-windows...
Could not locate cached archive: C:\Users\jinmhu.FAREAST\AppData\Local\vcpkg\archives\ac\ac9e446dff5521aab0e7951267f75d780019535e.zip
-- Downloading https://ftp.gnu.org/gnu/libiconv/libiconv-1.16.tar.gz...
-- Extracting source C:/Users/jinmhu.FAREAST/Desktop/vcpkg/downloads/libiconv-1.16.tar.gz
-- Applying patch 0002-Config-for-MSVC.patch
-- Applying patch 0003-Add-export.patch
-- Using source at C:/Users/jinmhu.FAREAST/Desktop/vcpkg/buildtrees/libiconv/src/1.16-de9101a181.clean
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
-- Installing: C:/Users/jinmhu.FAREAST/Desktop/vcpkg/packages/libiconv_x64-windows/share/libiconv/copyright
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: C:\Users\jinmhu.FAREAST\AppData\Local\vcpkg\archives\ac\ac9e446dff5521aab0e7951267f75d780019535e.zip
Building package libiconv[core]:x64-windows... done
Installing package libiconv[core]:x64-windows...
Installing package libiconv[core]:x64-windows... done
Elapsed time for package libiconv:x64-windows: 11.08 s
Starting package 4/5: liblzma:x64-windows
Building package liblzma[core]:x64-windows...
Could not locate cached archive: C:\Users\jinmhu.FAREAST\AppData\Local\vcpkg\archives\ad\ad30bf186dbd304becf7a06d3f6110f404de0233.zip
-- Downloading https://github.com/xz-mirror/xz/archive/v5.2.5.tar.gz...
-- Extracting source C:/Users/jinmhu.FAREAST/Desktop/vcpkg/downloads/xz-mirror-xz-v5.2.5.tar.gz
-- Applying patch enable-uwp-builds.patch
-- Using source at C:/Users/jinmhu.FAREAST/Desktop/vcpkg/buildtrees/liblzma/src/v5.2.5-19fec698ad.clean
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
-- Installing: C:/Users/jinmhu.FAREAST/Desktop/vcpkg/packages/liblzma_x64-windows/share/liblzma/copyright
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: C:\Users\jinmhu.FAREAST\AppData\Local\vcpkg\archives\ad\ad30bf186dbd304becf7a06d3f6110f404de0233.zip
Building package liblzma[core]:x64-windows... done
Installing package liblzma[core]:x64-windows...
Installing package liblzma[core]:x64-windows... done
Elapsed time for package liblzma:x64-windows: 13.39 s
Starting package 5/5: libxml2:x64-windows
Building package libxml2[core]:x64-windows...
Could not locate cached archive: C:\Users\jinmhu.FAREAST\AppData\Local\vcpkg\archives\ee\eef8f2197141d6d85702a3ee707a5071d5607c54.zip
-- Downloading https://github.com/GNOME/libxml2/archive/v2.9.10.tar.gz...
-- Extracting source C:/Users/jinmhu.FAREAST/Desktop/vcpkg/downloads/GNOME-libxml2-v2.9.10.tar.gz
-- Applying patch RemoveIncludeFromWindowsRcFile.patch
-- Using source at C:/Users/jinmhu.FAREAST/Desktop/vcpkg/buildtrees/libxml2/src/v2.9.10-073f92c335.clean
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
-- Using msys root at C:/Users/jinmhu.FAREAST/Desktop/vcpkg/downloads/tools/msys2/c809757c94447846
-- Fixing pkgconfig file: C:/Users/jinmhu.FAREAST/Desktop/vcpkg/packages/libxml2_x64-windows/lib/pkgconfig/libxml-2.0.pc
-- Fixing pkgconfig file: C:/Users/jinmhu.FAREAST/Desktop/vcpkg/packages/libxml2_x64-windows/debug/lib/pkgconfig/libxml-2.0.pc
-- Installing: C:/Users/jinmhu.FAREAST/Desktop/vcpkg/packages/libxml2_x64-windows/share/libxml2/copyright
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: C:\Users\jinmhu.FAREAST\AppData\Local\vcpkg\archives\ee\eef8f2197141d6d85702a3ee707a5071d5607c54.zip
Building package libxml2[core]:x64-windows... done
Installing package libxml2[core]:x64-windows...
Installing package libxml2[core]:x64-windows... done
Elapsed time for package libxml2:x64-windows: 18.27 s

Total elapsed time: 2.905 min

The package curl:x64-windows provides CMake targets:

    find_package(CURL CONFIG REQUIRED)
    target_link_libraries(main PRIVATE CURL::libcurl)

The package libxml2 is compatible with built-in CMake targets:

    find_package(LibXml2 REQUIRED)
    target_include_directories(main PRIVATE ${LIBXML2_INCLUDE_DIR})
    target_link_libraries(main PRIVATE ${LIBXML2_LIBRARIES})
```

This is evidence that proves command `vcpkg.exe install libxml2:x64-windows curl:x64-windows` actually works. Can you provide any evidence like that to prove your point?

Reply:
Failed proof.

```
PS E:\dev\src\cpp\vcpkg> vcpkg list
curl:x64-windows                                   7.68.0-3         A library for transferring data with URLs
curl[non-http]:x64-windows                                          Enables protocols beyond HTTP/HTTPS/HTTP2
curl[ssl]:x64-windows                                               Default SSL backend
curl[sspi]:x64-windows                                              SSPI support
curl[winssl]:x64-windows                                            SSL support (Secure Channel / "WinSSL")
libiconv:x64-windows                               1.16-1           GNU Unicode text conversion
libiconv:x64-windows-static                        1.16-1           GNU Unicode text conversion
liblzma:x64-windows                                5.2.4-5          Compression library with an API similar to that ...
liblzma:x64-windows-static                         5.2.4-5          Compression library with an API similar to that ...
libxml2:x64-windows                                2.9.9-5          Libxml2 is the XML C parser and toolkit develope...
zlib:x64-windows                                   1.2.11-6         A compression library
zlib:x64-windows-static                            1.2.11-6         A compression library

>------ Build All started: Project: azure-sdk-for-cpp, Configuration: x64-DebugWithTests ------
  [0/1] "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" --regenerate-during-build -SE:\dev\src\cpp\azure-sdk-for-cpp -BE:\dev\src\cpp\azure-sdk-for-cpp\out\build\x64-DebugWithTests
  -- Could NOT find CURL (missing: CURL_DIR)
  CMake Error at C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  -- Configuring incomplete, errors occurred!
  See also "E:/dev/src/cpp/azure-sdk-for-cpp/out/build/x64-DebugWithTests/CMakeFiles/CMakeOutput.log".
  See also "E:/dev/src/cpp/azure-sdk-for-cpp/out/build/x64-DebugWithTests/CMakeFiles/CMakeError.log".
    Could NOT find CURL (missing: CURL_LIBRARY CURL_INCLUDE_DIR) (Required is
    at least version "7.44")
  Call Stack (most recent call first):
    C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
    C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.18/Modules/FindCURL.cmake:169 (find_package_handle_standard_args)
    E:/dev/src/cpp/vcpkg/scripts/buildsystems/vcpkg.cmake:329 (_find_package)
    sdk/core/azure-core/CMakeLists.txt:16 (find_package)
  
  
  FAILED: build.ninja 
  "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" --regenerate-during-build -SE:\dev\src\cpp\azure-sdk-for-cpp -BE:\dev\src\cpp\azure-sdk-for-cpp\out\build\x64-DebugWithTests
E:\dev\src\cpp\azure-sdk-for-cpp\out\build\x64-DebugWithTests\ninja : error : rebuilding 'build.ninja': subcommand failed

Build All failed.
```
Success proof
```
PS E:\dev\src\cpp\vcpkg> vcpkg list
curl:x64-windows-static                            7.68.0-3         A library for transferring data with URLs
curl[non-http]:x64-windows-static                                   Enables protocols beyond HTTP/HTTPS/HTTP2
curl[ssl]:x64-windows-static                                        Default SSL backend
curl[sspi]:x64-windows-static                                       SSPI support
curl[winssl]:x64-windows-static                                     SSL support (Secure Channel / "WinSSL")
libiconv:x64-windows                               1.16-1           GNU Unicode text conversion
libiconv:x64-windows-static                        1.16-1           GNU Unicode text conversion
liblzma:x64-windows                                5.2.4-5          Compression library with an API similar to that ...
liblzma:x64-windows-static                         5.2.4-5          Compression library with an API similar to that ...
libxml2:x64-windows-static                         2.9.9-5          Libxml2 is the XML C parser and toolkit develope...
zlib:x64-windows                                   1.2.11-6         A compression library
zlib:x64-windows-static                            1.2.11-6         A compression library

1> CMake generation started for configuration: 'x64-DebugWithTests'.
1> Found and using vcpkg toolchain file (E:/dev/src/cpp/vcpkg/scripts/buildsystems/vcpkg.cmake).
1> Command line: "cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\ENTERPRISE\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="E:\dev\src\cpp\azure-sdk-for-cpp\out\install\x64-DebugWithTests" -DVCPKG_TARGET_TRIPLET:STRING="x64-windows-static" -DCMAKE_CXX_COMPILER:FILEPATH="C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/cl.exe" -DINSTALL_GTEST=OFF -DBUILD_TESTING=ON -DBUILD_TRANSPORT_CURL=ON -DBUILD_STORAGE_SAMPLES=ON -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\ENTERPRISE\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" -DCMAKE_TOOLCHAIN_FILE="E:/dev/src/cpp/vcpkg/scripts/buildsystems/vcpkg.cmake" "E:\dev\src\cpp\azure-sdk-for-cpp" 2>&1"
1> Working directory: E:\dev\src\cpp\azure-sdk-for-cpp\out\build\x64-DebugWithTests
1> [CMake] -- Found ZLIB: optimized;E:/dev/src/cpp/vcpkg/installed/x64-windows-static/lib/zlib.lib;debug;E:/dev/src/cpp/vcpkg/installed/x64-windows-static/debug/lib/zlibd.lib (found suitable version "1.2.11", minimum required is "1") 
1> [CMake] Libcurl version 7.68.0-DEV
1> [CMake] -- Retrieving version from E:/dev/src/cpp/azure-sdk-for-cpp/sdk/core/azure-core/inc/azure/core/version.hpp
1> [CMake] -- VERSION_MAJOR 1
1> [CMake] -- VERSION_MINOR 0
1> [CMake] -- VERSION_PATCH 0
1> [CMake] -- VERSION_PRERELEASE beta.4
1> [CMake] -- AZ_LIBRARY_VERSION 
1> [CMake] -- Retrieving version from E:/dev/src/cpp/azure-sdk-for-cpp/sdk/identity/azure-identity/inc/azure/identity/version.hpp
1> [CMake] -- VERSION_MAJOR 1
1> [CMake] -- VERSION_MINOR 0
1> [CMake] -- VERSION_PATCH 0
1> [CMake] -- VERSION_PRERELEASE beta.2
1> [CMake] -- AZ_LIBRARY_VERSION 
1> [CMake] -- Found ZLIB: optimized;E:/dev/src/cpp/vcpkg/installed/x64-windows-static/lib/zlib.lib;debug;E:/dev/src/cpp/vcpkg/installed/x64-windows-static/debug/lib/zlibd.lib (found version "1.2.11") 
1> [CMake] -- Retrieving version from E:/dev/src/cpp/azure-sdk-for-cpp/sdk/storage/azure-storage-common/inc/azure/storage/common/version.hpp
1> [CMake] -- VERSION_MAJOR 12
1> [CMake] -- VERSION_MINOR 0
1> [CMake] -- VERSION_PATCH 0
1> [CMake] -- VERSION_PRERELEASE beta.4
1> [CMake] -- AZ_LIBRARY_VERSION 
1> [CMake] -- Retrieving version from E:/dev/src/cpp/azure-sdk-for-cpp/sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/version.hpp
1> [CMake] -- VERSION_MAJOR 12
1> [CMake] -- VERSION_MINOR 0
1> [CMake] -- VERSION_PATCH 0
1> [CMake] -- VERSION_PRERELEASE beta.4
1> [CMake] -- AZ_LIBRARY_VERSION 
1> [CMake] -- Retrieving version from E:/dev/src/cpp/azure-sdk-for-cpp/sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/version.hpp
1> [CMake] -- VERSION_MAJOR 12
1> [CMake] -- VERSION_MINOR 0
1> [CMake] -- VERSION_PATCH 0
1> [CMake] -- VERSION_PRERELEASE beta.4
1> [CMake] -- AZ_LIBRARY_VERSION 
1> [CMake] -- Retrieving version from E:/dev/src/cpp/azure-sdk-for-cpp/sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/version.hpp
1> [CMake] -- VERSION_MAJOR 12
1> [CMake] -- VERSION_MINOR 0
1> [CMake] -- VERSION_PATCH 0
1> [CMake] -- VERSION_PRERELEASE beta.4
1> [CMake] -- AZ_LIBRARY_VERSION 
1> [CMake] -- Retrieving version from E:/dev/src/cpp/azure-sdk-for-cpp/sdk/template/azure-template/inc/azure/template/version.hpp
1> [CMake] -- VERSION_MAJOR 1
1> [CMake] -- VERSION_MINOR 0
1> [CMake] -- VERSION_PATCH 0
1> [CMake] -- VERSION_PRERELEASE beta.8
1> [CMake] -- AZ_LIBRARY_VERSION 
1> [CMake] -- Configuring done
1> [CMake] -- Generating done
1> [CMake] -- Build files have been written to: E:/dev/src/cpp/azure-sdk-for-cpp/out/build/x64-DebugWithTests
1> Extracted CMake variables.
1> Extracted source files and headers.
1> Extracted code model.
1> Extracted includes paths.
1> CMake generation finished.
```

Reply:
let's put side whether this issue is caused by some misconfiguration in your environment.

What do you suggest us do in this PR? Are you suggesting that we should drop the support for dynamic library dependencies? Is so, I think this is a big decision that neither you nor I can make, we should engage @vinjiang or C++ working group.

Reply:
> let's put side whether this issue is caused by some misconfiguration in your environment.

Let's not put aside this claim. Can you please provide evidence of what misconfiguration would cause this issue?

> > > > > Still need some justification for this change
> > > > 
> > > > 
> > > > Justification provided.
> > > 
> > > 
> > > What do you suggest us do in this PR? Are you suggesting that we should drop the support for dynamic library dependencies? Is so, I think this is a big decision that neither you nor I can make, we should engage @vinjiang or C++ working group.
> > 
> > 
> > No, not at all. I suggest that what we wrote in Readme should at least work with the default setting, that is all. Customers shouldn't come back to us and tell us that they followed the Readme only to find there is some 'package not found' error.
> 
> Then how are we going to follow up on that issue caused by dynamic library? Do you have any issue tracking that?

Reply:
> Can you please provide evidence of what misconfiguration would cause this issue?

I cannot reproduce that issue in my environment with dynamic library. I've been working with dynamic library since the very beginning of this project. It's been working for me in the last year. Do you think that's a valid envidence?

Reply:
> > Can you please provide evidence of what misconfiguration would cause this issue?
> 
> I cannot reproduce that issue in my environment with dynamic library. I've been working with dynamic library since the very beginning of this project. It's been working for me in the last year. Do you think that's a valid evidence?

Did you at least try [open with folder](https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/storage#use-visual-studios-open-by-folder-feature) approach to build this library?

Reply:
This is caused by mismatch of triplets in [CMakeSettings.json](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CMakeSettings.json) and README. My question is why not change the one in CMakeSettings.json?